src/Chapter4.hs

```diff
@@ -611,6 +650,15 @@ Implement the 'Monad' instance for our lists.
   maybe a few) to flatten lists of lists to a single list.
 -}
 
+flattenList :: List (List a) -> List a
+flattenList Empty = Empty
+flattenList (Cons x (Cons y rest)) = x <> y <> (flattenList rest)
+
+instance Monad List where
+    (>>=) :: List a -> (a -> List b) -> List b
+    (>>=) Empty _ = Empty
```

Comment:
This case is redundant here because both `flattenList` and `fmap` return `Empty` on the `Empty` constructor, so you don't need to handle this explicitly. See, how it's enough to handle all patterns in basic functions, and then you can compose them easily :slightly_smiling_face:

Reply:
Man, you are good at spotting things like that, I will try to be more thoughtful of this. It looks like a common theme in Haskell.